dspy/modeling/templates/json.py

```diff
@@ -0,0 +1,87 @@
+
+import json
+
+from dspy import Example, Signature
+from dspy.primitives.prompt import Prompt
+
+from .base import BaseTemplate
+
+
+class JSONTemplate(BaseTemplate):
+    def generate(self, signature: Signature, example: Example) -> Prompt:
+
+        prompt_spans = []
+
+        # Start by getting the instructions
+        prompt_spans.append(signature.instructions)
+
+        # Generate the Guidelines
+        prompt_spans.append(self._guidelines(signature, example))
+
+        # Generate spans for all the demos
+        for demo in example.demos:
+            prompt_spans.append(self._example_span(signature, demo))
+
+        # Generate span for the active example
+        prompt_spans.append(self._example_span(signature, example))
+
+        content = "\n\n--\n\n".join(prompt_spans)
+        return Prompt(content=content, messages=[{"role": "user", "content": content}])
```

Comment:
Following https://github.com/stanfordnlp/dspy/issues/662 this could also be
```
messages: [
{"from": "system", "message": guidance},
{"from": "user", "message": input0},
{"from": "agent", "message": output0},
{"from": "user", "message": input1},
{"from": "agent", "message": output2},
{"from": "user", "message": input2},
]
```

But maybe we should just make a separate JSONChatTemplate?